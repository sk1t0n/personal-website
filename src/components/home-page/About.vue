<template>
  <section id="about">
    <h2>{{ $t('about.word1') }} <span>{{ $t('about.word2') }}</span></h2>
    <div class="row">
      <q-img :src="photoUrl" />
      <div>
        <p>{{ $t('about.p1') }}</p>
        <p>{{ $t('about.p2') }}</p>
        <p>{{ $t('about.p3') }}</p>
        <p>{{ $t('about.p4') }}</p>
        <p>{{ $t('about.p5') }}</p>
        <p>{{ $t('about.p6') }}</p>
      </div>
    </div>
  </section>
</template>

<script setup>
const photoUrl = 'images/photo.jpeg'
</script>

<style lang="scss">
#about {
  .row {
    margin-top: 20px;
    justify-content: center;

    .q-img {
      max-width: 256px;
      max-height: 256px;
    }

    div {
      padding-top: 0.6em;
    }

    p {
      font-size: 1em;
      margin-top: 10px;
      margin-bottom: 10px;
    }
  }
}

@media screen and (max-width: 767px) {
  #about {
    .row {
      flex-direction: column;
      justify-content: center;
      margin-left: 2em;
      margin-right: 2em;

      .q-img,
      div {
        margin: 0 auto;
      }

      p:first-child {
        margin-top: 20px;
      }
    }
  }
}

@media screen and (min-width: 768px) {
  #about {
    .row {
      .q-img {
        margin-right: 30px;
      }

      p:first-child {
        margin-top: 17px;
      }
    }
  }
}
</style>
