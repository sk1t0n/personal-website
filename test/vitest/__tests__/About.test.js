import { installQuasar } from '@quasar/quasar-app-extension-testing-unit-vitest'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import About from 'src/components/home-page/About.vue'
import translations from 'src/i18n/en-US'

installQuasar()

describe('About.vue', () => {
  const wrapper = mount(About, {
    global: {
      mocks: {
        $t: (msg) => {
          const [k1, k2] = msg.split('.')
          return translations[k1][k2]
        },
      },
    },
  })

  it('header has correct text', () => {
    const text = 'About Me'
    expect(wrapper.find('h2').text()).toBe(text)
  })

  it('q-img has correct url', () => {
    const text = 'images/photo.jpeg'
    expect(wrapper.vm.photoUrl).toBe(text)
  })

  it('correct snapshot', () => {
    expect(wrapper.text()).toMatchSnapshot()
  })
})
