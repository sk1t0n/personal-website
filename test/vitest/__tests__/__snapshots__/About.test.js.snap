// Vitest Snapshot v1

exports[`About.vue > correct snapshot 1`] = `"About MeFullstack DeveloperProgramming languages: PHP 8.1, JavaScript (ES6 and later)Backend: SymfonyDatabases: PostgreSQL, MongoDBFrontend: Vue.js, Vuex, Quasar Framework, Buefy, Nuxt.jsDevOps: Docker, Docker Compose"`;
